p2p/node.go

```diff
@@ -167,12 +181,12 @@ func (n *QriNode) Connect() (err error) {
 		n.Online = true
 		go n.echoMessages()
 	}
-	return nil
+	return n.startOnlineServices()
```

Comment:
it's important that `n.startOnlineServices` is only called when tansitioning from offline to online state. To do that we should move `return n.startOnlineServices()` up and into the `if !n.Online {` closure, and restore the `return nil` that should happen otherwise.

Come to think of it, it'd be great if we removed the giant `if !n.Online` and instead wrote:
```go
if n.Online {
  return nil 
}

// Then to all the go-online stuff here
```